File: BankKata/BankCore/Models/Transaction.cs
namespace BankCore.Models
{
    public class Transaction
    {
        private readonly int _amount;
        private readonly string _date;

        public Transaction(string date, int amount)
        {
            _date = date;
            _amount = amount;
        }

        public string Date
        {
            get { return _date; }
        }

        public int Amount
        {
            get { return _amount; }
        }

        public override bool Equals(object obj)
        {
            if (this == obj)
                return true;
            if (obj == null || obj.GetType() != GetType())
                return false;

            Transaction that = (Transaction)obj;

            if (_date != that._date || _amount != that._amount)
                return false;

            return true;
        }
    }
    
}

File: BankKata/BankCoreVBNetEdition/Transaction.vb
Public Class Transaction
    Private _amount As Integer
    Private _date As String

    Public Sub New([date] As String, amount As Integer)
        _date = [date]
        _amount = amount
    End Sub


    Public ReadOnly Property Amount As Integer
        Get
            Return _amount
        End Get
    End Property

    Public ReadOnly Property [Date] As String
        Get
            Return _date
        End Get
    End Property

    Public Overrides Function Equals(obj As Object) As Boolean
        If (obj Is Me) Then
            Return True
        End If
        If (obj Is Nothing Or TypeOf obj IsNot Transaction) Then
            Return False
        End If

        Dim that As Transaction = CType(obj, Transaction)

        If (_date <> that._date Or _amount <> that._amount) Then
            Return False
        End If

        Return True

    End Function


End Class
